[18.0][MIG] stock_picking_volume: Migration to 18.0

henrybackman opened the pull request:
This PR is building on top of the earlier migration PR by @arantxa-s73 
- https://github.com/OCA/stock-logistics-warehouse/pull/2258

Changes done:
- Moved `readonly` definitions to views
- Handled case when `product_product:_get_volume_for_qty` is called with empty recordset

Ported changes from
- https://github.com/OCA/stock-logistics-warehouse/pull/2282

lmignon committed:
[ADD] stock_picking_volume: New addon to compute and store the volume of a picking

lmignon committed:
[IMP] stock_picking_volume: Avoid useless recompute

Makes computed fields readonly in state 'done' or 'cancel' to avoid triggering recompute when these values are useless

lmignon committed:
[IMP] stock_picking_volume: Speedup installation

Init computed fields from plain sql queries to speedup install on large database

mt-software-de committed:
[FIX] stock_picking_volume: stock.picking _compute_volume

Calculate the volume only from not canceled moves

mt-software-de committed:
[IMP] stock_picking_volume: Move _get_volume_for_qty to product.product

mymage committed:
Added translation using Weblate (Italian)

ferran-S73 committed:
[FIX] stock_pikcing_volume: ambiguous column

RabbitJon-S73 committed:
[MIG] stock_picking_volume: Migration to 17.0

RabbitJon-S73 committed:
[STYLE] stock_picking_volume: pre-commit

Commit:
[UPD] Update stock_picking_volume.pot

OCA-git-bot committed:
[BOT] post-merge updates

sebalix committed:
stock_picking_volume: pre_init_hook, replace column type from 'numeric' to 'double precision'

As no digits parameters are provided on `<stock.move>.volume' and
'<stock.picking>.volume' fields, the 'double precision' type is used
instead of 'numeric'.

This avoids time consuming SQL requests 'ALTER TABLE...' at module
install.

sebalix committed:
[FIX] stock_pikcing_volume: pre_init_hook, ambiguous columns

joelkm committed:
[IMP] stock_picking_volume: add volume and uom to tree view

arantxa-s73 committed:
[IMP] stock_picking_volume: pre-commit auto fixes

arantxa-s73 committed:
[MIG] stock_picking_volume: Migration to 18.0

henrybackman committed:
[MIG] stock_picking_volume: Migration to 18.0

sebalix commented:
/ocabot migration stock_picking_volume

sebalix commented on stock_picking_volume/README.rst:
```diff
@@ -0,0 +1,121 @@
+
+|badge1| |badge2| |badge3| |badge4| |badge5|
+
+For some processes, you need to know the volume of your stock picking.
+This could be useful for example to know the device to use to process
+it. An other use case could be to know how many trucks you need to
+deliver your products.
+
+By default, the delivery module from Odoo will compute the picking
+weight. Unfortunately, it will not compute the volume. This module will
+add the volume computation to the picking and to the stock move. This
+information will be stored in the database and will be available on the
+picking form view and on the stock move form view.
```
Taking the opportunity to remove some technical terms:
```suggestion
For some processes, you need to know the volume of your stock transfers.
This could be useful for example to know the device to use to process
it. An other use case could be to know how many trucks you need to
deliver your products.

By default, the delivery module from Odoo will compute the transfer
weight. Unfortunately, it will not compute the volume. This module will
add the volume computation to the stock transfer and to the moves. This
information will be stored in the database and will be available on the
transfer and move form views.
```

henrybackman replied:
Done

sebalix reviewed (approved):
LG, small remarks

sebalix commented:
Just saw there was already a PR open for that module there:
- https://github.com/OCA/stock-logistics-warehouse/pull/2258

Can you please compare its migration with yours and state?

sebalix commented:
I also open this PR:
- https://github.com/OCA/stock-logistics-warehouse/pull/2282
Would be nice to cherry-pick its commits

henrybackman commented:
> Just saw there was already a PR open for that module there:
> 
> * [[18.0][MIG] stock_picking_volume: Migration to 18.0  #2258](https://github.com/OCA/stock-logistics-warehouse/pull/2258)
> 
> Can you please compare its migration with yours and state?

Added my migration changes on top of that migration PR

> I also open this PR:
> 
> https://github.com/OCA/stock-logistics-warehouse/pull/2282
> Would be nice to cherry-pick its commits

Added those commits to this PR as well

grindtildeath reviewed (approved):
Looks good

jbaudoux commented:
/ocabot merge nobump
